tests/cephadm/test_cephadm_staggered_upgrade.py

```diff
@@ -0,0 +1,123 @@
+import json
+
+from ceph.ceph_admin.orch import Orch
+from ceph.waiter import WaitUntil
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class StaggeredUpgradeError(Exception):
+    pass
+
+
+def check_upgrade_status(node):
+    """Check upgrade status
+    Args:
+        node: installer node
+    """
+    timeout, interval = 600, 5
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        out = CephAdm(node).ceph.orch.upgrade.status()
+        log.info(f"upgrade status: {out}")
+        if not json.loads(out)["in_progress"]:
+            break
+        if w.expired:
+            raise StaggeredUpgradeError("Cluster upgrade not completed")
+
+
+def run(ceph_cluster, **kw):
+    """Staggered upgrade
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test
+        kw: test data
+        e.g:
+        test:
+            name: Staggered upgrade with daemon types mgr,mon
+            desc: Staggered upgrade with daemon types mgr,mon
+            module: test_cephadm_staggered_upgrade.py
+            polarion-id: CEPH-83575554
+            config:
+                action: "daemon_types"
+                osd_flags:
+                - noout
+                - noscrub
+                - nodeep-scrub
+                daemon_types: mgr,mon
+    """
+    config = kw.get("config")
+    osd_flags = config.get("osd_flags")
+    target_image = config.get("container_image")
+    action = config.get("action")
+    node = ceph_cluster.get_nodes(role="mon")[0]
+    orch = Orch(cluster=ceph_cluster, **config)
+    # Check cluster health before upgrade
+    if orch.cluster.check_health(rhbuild=config.get("rhbuild"), client=orch.installer):
```

Comment:
One suggestion, can we use the existing cli method health here?
ceph_health = CephAdm(installer_node).ceph.health(), so that the orch class need not be imported here?

Reply:
Thanks @pranavprakash20 
Updated